src/main/java/calculator/Calculator.java

```diff
@@ -0,0 +1,61 @@
+package calculator;
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Calculator {
+
+	private static final String SPLIT_REGEX = "[,:]";
+	private static final String CUSTOM_SPLIT_REGEX = "//(.)\\n(.*)";
+	private static final int DELIMITER_NUMBER = 1;
+	private static final int INPUT_GROUP = 2;
+
+	public static int splitAndSum(String input) {
+		if (isEmpty(input)) {
+			return 0; // 빈 문자열 또는 null 값을 입력할 경우 0을 반환해야 한다.
+		}
+		return getSum(split(input));
+	}
+
+	private static int getSum(String[] inputs) {
+		for (String input : inputs) {
+			checkInputIsNegative(input);
+		}
+
+		return Arrays.stream(inputs)
+			.mapToInt(Integer::parseInt)
+			.reduce(0, Integer::sum);
+	}
+
+	private static void checkInputIsNegative(String input) {
+		if (getParseInt(input) < 0) {
+			throw new RuntimeException("숫자 이외의 값 또는 음수는 계산할 수 없습니다.");
+		}
+	}
+
+	private static int getParseInt(String input) {
+		return Integer.parseInt(input);
+	}
+
+	private static String[] split(String input) {
+		Matcher matcher = getMatcher(input);
+		if (isCustom(matcher)) { // 커스텀 구분자를 사용한다면
+			String customDelimiter = matcher.group(DELIMITER_NUMBER);
+			return matcher.group(INPUT_GROUP).split(customDelimiter);
+		}
+		return input.split(SPLIT_REGEX);
+	}
+
+	private static boolean isCustom(Matcher matcher) {
+		return matcher.find();
+	}
+
+	private static Matcher getMatcher(String input) {
+		return Pattern.compile(CUSTOM_SPLIT_REGEX).matcher(input);
+	}
```

Comment:
정규식 일치 여부를 확인할 때 메서드 내에서 Patterm.compile()을 그냥 사용하게 되면 메서드가 호출될 때마다 컴파일이 발생해 성능에 영향을 미쳐요😭. 
아래와 같이 바꾸는게 좋아요!
```suggestion
private static final Pattern COMPILED_CUSTOM_SPLIT_REGEX = Pattern.compile(CUSTOM_SPLIT_REGEX);

private static Matcher getMatcher(String input) {
	return COMPILED_CUSTOM_SPLIT_REGEX.matcher(input);
}
```

Reply:
기존 코드는 getMatcher 메서드가 호출될 때 마다 Pattern.compile 을 호출하네요..😭

https://medium.com/webeveloper/%EC%9E%90%EB%B0%94-%EC%98%AC%EB%B0%94%EB%A5%B4%EA%B2%8C-%EC%A0%95%EA%B7%9C%EC%8B%9D-%ED%8C%A8%ED%84%B4-%EC%82%AC%EC%9A%A9%ED%95%98%EA%B8%B0-c2479c33ab8b

패턴을 여러 번 사용하는 경우 한 번 컴파일한 후 재사용
* 매번 이 메서드를 호출하는 것보다 효율적입니다.

라고 하네요! 수정 완료했습니다.😄

Reply:
수정한 부분 확인했습니다~. 그런대 상수의 경우 대문자로 표기하는게 컨벤션 아니였나요? pattern이 소문자로 되어있네요...

Reply:
아 그렇네요😅 대문자 컨벤션을 지켜 수정했습니다!